### DISCUSSION TITLE
Not understanding block_lists

### DISCUSSION THREAD

**Discussion by cstradtman** [mitmproxy/mitmproxy#5346] [Q&A]
I've been trying to use mitm to block (return 404s) for certain domains.  I'm obviously not understanding something.  
if I wanted to block 
a.example.com
b.example.com
c.example.com
would
```
block_list:
- /~d example.com/404
```
in config.yaml
do the job?  (because that's what I'm trying and it doesn't seem to have any effect)
or is this the sort of thing I need to do my on addon for?
Does using allow_hosts somehow trip it up?

thanks,
Chris

**Comment by mhils** [comment]
What you are doing looks generally correct. Some examples:
```shell
mitmdump --set "block_list=/~d example.com/404"
curl -x localhost:8080 -k https://example.com  # properly returns a 404

mitmdump --set "block_list=/~d example.com/404"
curl -x localhost:8080 -k https://a.example.com  # 502 because mitmproxy cannot fetch the certificate.

# connection_strategy=lazy -> handle "request" hook before ever connecting upstream
mitmdump --set "block_list=/~d example.com/404" --set connection_strategy=lazy  
mcurl https://a.example.com  # now returns a 404 as well
```

It should not matter if you use `config.yaml` or command-line arguments, both should have the same effect. You can verify that by opening the options menu in mitmproxy (press `O`).

**Comment by cstradtman** [comment]
I actually figured out what was going on ( a little sleep helps ;-) )
Since I was using allow_hosts I was just entering the host that I wanted to do the man in the middle on. for example

```
allow_hosts:
- mitmexample.com

block_list:
- /~d example.com/404
```
as soon as I change  this to 

```
allow_hosts:
- mitmexample.com
- example.com

block_list:
- /~d example.com/404
```
it started working as expected.

Thanks for the response
Chris
